### PR TITLE
Just a test of scatterplot - do not merge

### PR DESCRIPTION
**Pull request by adrianmroz-allegro**

### CODE REVIEW

File: src/client/visualizations/scatterplot/scatterplot.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  setPointHover = (datum: Datum): void =>
+    this.setState({ hoveredPoint: datum });
+
+  resetPointHover = (): void =>
+    this.setState({ hoveredPoint: null });
+
+  render() {
+    const { data, essence, stage } = this.props;
+    const splitKey = essence.splits.splits.first().toKey();
```
Comment (mkuthan) [inline src/client/visualizations/scatterplot/scatterplot.tsx:63]: Should we have some checks for nulls in the chain here?

File: src/client/visualizations/scatterplot/scatterplot.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    const { xTicks, yTicks, xScale, yScale, xSeries, ySeries, plottingStage, scatterplotData } = this.getPlottingData(data, essence, stage);
+
+    const rightXAxisLabelPosition = stage.width - (plottingStage.width + plottingStage.x);
```
Comment (mkuthan) [inline src/client/visualizations/scatterplot/scatterplot.tsx:67]: Candidate for extracted method with unit test.

File: src/client/visualizations/scatterplot/tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+const TOOLTIP_OFFSET_X = 100;
+
+export const Tooltip: React.SFC<TooltipProps> = ({ datum, stage, xSeries, ySeries, xScale, yScale, splitKey }) => {
+  if (!Boolean(datum)) return null;
+
+  const title = datum[splitKey] as string;
+  const xValue = xSeries.selectValue(datum);
+  const yValue = ySeries.selectValue(datum);
+
+  return <TooltipWithinStage top={yScale(yValue) + TOOLTIP_OFFSET_Y} left={xScale(xValue) + TOOLTIP_OFFSET_X} stage={stage}>
```
Comment (mkuthan) [inline src/client/visualizations/scatterplot/tooltip.tsx:47]: Extract calculation, xValue and yValue could be defined within offsets in lines 44/45.

File: src/client/visualizations/scatterplot/x-axis.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  stage: Stage;
+  ticks: number[];
+  tickSize: number;
+  scale: LinearScale;
+  formatter: Unary<number, string>;
+}
+
+export const XAxis: React.SFC<XAxisProps> = ({ stage, ticks, scale, formatter, tickSize }) => {
+  const labelY = tickSize + TEXT_OFFSET_X;
+  const linePositionY = roundToHalfPx(0);
```
Comment (mkuthan) [inline src/client/visualizations/scatterplot/x-axis.tsx:37]: What does round(0) mean?

**Review by mkuthan — COMMENTED** [review]
I'm not sure that I commented correct PR but anyway - a few comments.

File: src/client/visualizations/scatterplot/scatterplot.tsx (hunk 1 above)
Comment (adrianmroz-allegro) [reply src/client/visualizations/scatterplot/scatterplot.tsx:63]: Nope. Resolvers run earlier and they enforce that we have exactly one split.

File: src/client/visualizations/scatterplot/x-axis.tsx (hunk 4 above)
Comment (adrianmroz-allegro) [reply src/client/visualizations/scatterplot/x-axis.tsx:37]: It is round to half, not just round. Result will be always 0.5, but method call show intent.

File: src/client/visualizations/scatterplot/scatterplot.tsx (hunk 2 above)
Comment (kzadurska) [reply src/client/visualizations/scatterplot/scatterplot.tsx:67]: Will extract, but not on this branch.

**Comment by kzadurska** [comment]
> I'm not sure that I commented correct PR but anyway - a few comments.

@mkuthan Nope, this is just a PR to test the branch deployment. I will add changes in another PR to the proper feature branch. 
Should have included you in the review in the first place, my bad 😬

File: src/client/visualizations/scatterplot/tooltip.tsx (hunk 3 above)
Comment (kzadurska) [reply src/client/visualizations/scatterplot/tooltip.tsx:47]: Will extract, but not on this branch.

Comment (kzadurska) [reply src/client/visualizations/scatterplot/tooltip.tsx:47]: https://github.com/allegro/turnilo/pull/853/files#r814671010

File: src/client/visualizations/scatterplot/scatterplot.tsx (hunk 2 above)
Comment (kzadurska) [reply src/client/visualizations/scatterplot/scatterplot.tsx:67]: https://github.com/allegro/turnilo/pull/853/files#r814670493

**Comment by kzadurska** [comment]
Since this is just a test PR, closing. Moved the code review comments that need action to https://github.com/allegro/turnilo/pull/853/files

**Pull request closed without merging**